agent.py

```diff
@@ -45,6 +46,11 @@ def __init__( self, agentname, name, password, holon, token, args={}, flows=[] )
         # LHS it is an output. If self isn't present
         # then it is a forward (not processed LHS input is
         # forwarded directly to RHS output)
+        self.environment = None
+        for i in self.flows:
+            if ( i[0] == self.holon_name or i[1] == self.holon_name ) and not i[2] == None:
+                self.environment = { 'name': self.holon_name, 'io_name': i[2] }
+                break
```

Comment:
Could be a separate fn. This block checks if in `flows` there is a flow defined for communication with environment